api/activetigger/functions.py

```diff
@@ -41,10 +42,40 @@
     BertTokenizer,
     Trainer,
     TrainerCallback,
+    TrainerControl,
+    TrainerState,
     TrainingArguments,
 )
 
 
+class CustomLoggingCallback(TrainerCallback):
+    event: Optional[multiprocessing.synchronize.Event]
+    current_path: str
+    logger: Logger
+
+    def __init__(self, event, logger, current_path):
+        self.event = event
+        self.current_path = current_path
+        self.logger = logger
+
+    def on_step_end(
+        self,
+        args: TrainingArguments,
+        state: TrainerState,
+        control: TrainerControl,
+        **kwargs,
+    ):
+        self.logger.info(f"Step {state.global_step}")
+        progress_percentage = (state.global_step / state.max_steps) * 100
+        with open(self.current_path / "train/progress", "w") as f:
```

Comment:
I'm quasi sure that if `current_path` is a `str`, then `self.current_path / "train/progress"` will fail on the `/`, it is likely `current_path` is a `pathlib.Path`, 

And you want to do `(self.curent_path/"train"/"progress").write_text(data)` (also I'm not sure of the cross-platform behavior of having a `/`  in a string; it is possible, that `self.current_path / "train/progress"`  could create a file actually named 'train/progress'; but I think you want the `/` outside the string.

(I know the `/` things is from the original version ).


Also to catch these types of errors; later on you want to turn on [mypy](https://pypi.org/project/mypy/) in CI.